src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -0,0 +1,450 @@
+    """
+    def load_schema_module_list(self, yang_files):
+        for file in yang_files:
+            if (self.load_schema_module(file) == False):
+                return False
+        return True
+
+    """
+    load_schema_modules(): load all Yang model files in the directory
+    input:    yang_dir - the directory of the yang model files to be loaded
+    returns:  True - success   False - failed
+    """
+    def load_schema_modules(self, yang_dir):
+        py = glob.glob(yang_dir+"/*.yang")
+        for file in py:
+            if (self.load_schema_module(file) == False):
+                return False
+        return True
+
+    """
+    load_schema_modules_ctx(): load all Yang model files in the directory to context: ctx
+    input:    yang_dir,  context
+    returns:  context
+    """
+    def load_schema_modules_ctx(self, yang_dir):
+        ctx = self.ctx
+        py = glob.glob(yang_dir+"/*.yang")
+        for file in py:
+            ctx.parse_module_path(str(file), ly.LYS_IN_YANG)
+
+        return ctx
+
+    """
+    load_data_file(): load a Yang data json file
+    input:    data_file - the full path of the yang json data file to be loaded
+    returns:  True - success   False - failed
+    """
+    def load_data_file(self, data_file):
+       node = None
+       try:
+            node = self.ctx.parse_data_path(data_file, ly.LYD_JSON, ly.LYD_OPT_CONFIG | ly.LYD_OPT_STRICT)
+       except Exception as e:
+            print(e)
+            return False
+       self.root = node
+       return True
+
+    """
+    get module name from xpath
+    input:    xpath
+    returns:  module name
+    """
+    def get_module_name(self, xpath):
+        module_name = xpath.split(':')[0].strip('/')
+        return module_name
+
+    """
+    get_module(): get module object from Yang module name
+    input:   yang module name
+    returns: Schema_Node object
+    """
+    def get_module(self, module_name):
+        mod = self.ctx.get_module(module_name)
+        return mod
+
+    """
+    load_data_model(): load both Yang module file and data json file
+    input:   yang directory, yang file and data file
+    returns: (context, root)
+    """
+    def load_data_model (self, yang_dir, yang_file, data_file, output):
+        if (self.ctx is None):
+            self.ctx = ly.Context(yang_dir)
+
+        if self.module is  None:
+            module = self.ctx.load_module(yang_file)
+            self.module = module
+
+        try:
+            node = self.ctx.parse_data_path(data_file, ly.LYD_JSON, ly.LYD_OPT_CONFIG | ly.LYD_OPT_STRICT)
+        except Exception as e:
+            node = None
+            print(e)
+
+        self.root = node
+        return (self.ctx, node)
+
+    """
+    print_data_mem():  print the data tree
+    input:  option:  "JSON" or "XML"
+    """
+    def print_data_mem (self, option):
+        if (option == "JSON"):
+            mem = self.root.print_mem(ly.LYD_JSON, ly.LYP_WITHSIBLINGS | ly.LYP_FORMAT)
```

Comment:
This print correct xml format for yang model. I think we can file a bug against LIbyang.

Reply:
Will file a bug against libyang